src/complete.ts

```diff
@@ -15,27 +15,57 @@ function stripQuotes(name: string) {
   return quoted ? quoted[1] : name
 }
 
+function isIdentifier(state: EditorState, node: SyntaxNode) {
+  let value = stripQuotes(state.sliceDoc(node.from, node.to)).toLowerCase()
+
+  // Handle a case when a "public" schema is specified, which also counts as a SQL keyword
+  return node.name == "Identifier" || node.name == "QuotedIdentifier" || (node.name == "Keyword" && value == "public")  
+}
+
+function schemaBefore(state: EditorState, tree: SyntaxNode) {
+  let dot = tokenBefore(tree)
+
+  if (dot && dot.name == ".") {
+    let schema = tokenBefore(dot)
+
+    if (schema && isIdentifier(state, schema)) {
+      return schema
+    }
+  }
+
+  return null
+}
+
 function sourceContext(state: EditorState, startPos: number) {
   let pos = syntaxTree(state).resolveInner(startPos, -1)
   let empty = false
-  if (pos.name == "Identifier" || pos.name == "QuotedIdentifier") {
+  if (isIdentifier(state, pos)) {
     empty = false
     let parent = null
     let dot = tokenBefore(pos)
     if (dot && dot.name == ".") {
       let before = tokenBefore(dot)
-      if (before && before.name == "Identifier" || before.name == "QuotedIdentifier")
```

Comment:
I extracted this logic into `isIdentifier` function, so that the same `sourceContext` function supports autocomplete of tables under a schema without any modification to it. It was also repeated in 3 places, so I think it made sense to extract it either way.